cdap-ui/app/directives/validators/validator-factory.js

```diff
@@ -0,0 +1,120 @@
+     *    3. For each if-else tree:
+     *        a. Find the field name from the arguments
+     *        b. Create validation key '<validator object>.<validation function name>'
+     *        c. Add the rest of the arguments for the validation function (if any)
+     *        d. Push object to the fieldGroup
+     *        e. Check if the if block body contains another if-else block.
+     *            - If yes, call _addRule with operation true (means to treat the
+     *                next rule as AND operation)
+     *            - If no, check if alternate body contains another if-else block. If yes,
+     *                call _addRule with opertaion false (treat next rule as OR operation)
+     **/
+
+    var block = jsonTree.body[0].body.body;
+    var initialValidation;
+    for (var i = 0; i < block.length; i++) {
+      if (block[i].type === 'IfStatement') {
+        initialValidation = block[i];
+        break;
+      }
+    }
+
+    var fieldGroup = {};
+    _addRule(fieldGroup, initialValidation, functionMap, true);
+
+    return fieldGroup;
+  }
+
+  function _findFieldArgument(args) {
+    for (var j = 0; j < args.length; j++) {
+      if (args[j].type === 'MemberExpression') {
+        return args[j];
+      }
+    }
+  }
+
+  function _addRule (fieldGroup, rule, functionMap, operation) {
+    // Find validation rule argument
+    var field = _findFieldArgument(rule.test.arguments);
+
+    var fieldName = field.property.name;
+
+    // ie. coreValidator.maxLength
+    var validationKey = rule.test.callee.object.name + '.' + rule.test.callee.property.name;
+
+    if (!fieldGroup[fieldName]) {
+      fieldGroup[fieldName] = [];
+    }
+
+    var obj = {
+      fieldName: fieldName,
+      operation: operation,
+      validation: validationKey
+    };
+
+    // Adding the rest of validation function arguments (if any)
+    var argsValidationRule = functionMap[validationKey].arguments;
+    if (argsValidationRule.length > 1) {
+      obj.arguments = {};
+      angular.forEach(argsValidationRule, function (val, index) {
+        if (!val.startsWith('<field:')) {
+          var argValue = rule.test.arguments[index];
+          obj.arguments[val] = argValue.value;
+        }
+      });
+    }
+
+    fieldGroup[fieldName].push(obj);
+
+    // Recursive call
+    if (rule.consequent.body.length && rule.consequent.body[0].type === 'IfStatement') {
+      // Treat next rule as AND
+      _addRule(fieldGroup, rule.consequent.body[0], functionMap, true);
+    } else if (rule.alternate.body.length && rule.alternate.body[0].type === 'IfStatement') {
+      // Treat next rule as OR
+      _addRule(fieldGroup, rule.alternate.body[0], functionMap, false);
+    }
+  }
+
+  return {
+    initValidationFields: initValidationFields
+  };
+}
+
+angular.module(PKG.name + '.commons')
+  .factory('esprima', function ($window) {
+    /**
+     * esprima is being used to parse JS code to a JSON tree
+     **/
+
+    return $window.esprima;
```

Comment:
Can we add this in `app/lib/global-lib.js`. We could have one place to see where is some global library being injected into